### DISCUSSION TITLE
Difference between SimNIBS and Freesurfer segmentation volumes

### DISCUSSION THREAD

**Discussion by dimst23** [simnibs/simnibs#316] [9. Results and analysis]
Hello!

I want to extract some anatomical structure metrics (cortical thickness, volume among others) through freesurfer, but in order to run simulations in SimNIBS I need to have a charm segmented model.

In order to use the anatomical measures extracted from freesurfer in my analysis in conjunction with the electric fields from SimNIBS, I need to understand if the brain models are comparable.

I calculated the volume of both the freesurfer segmentation and charm segmentation using the `mri_label_volume` command of freesurfer and using the `mri/aseg.mgz` from the freesurfer segmentation and `segmentation/labeling.nii.gz` from charm.

I have found differences sometimes reaching as high as **30-50%**. Differences between the same volume label between the freesurfer and charm segmentation. I checked the segmentation quality for any distortions or evident difference but couldn't detect any. Below I attach a graph that shows the percentage difference for 6 sample subjects and as you see the results are variable. The names of the areas and the subsequent labels correspond to the `segmentation/labeling_LUT.txt`.

**charm settings**
- Default `settings.ini`
- Flags: `--fs-dir`, `--skipregisterT2`, `--noneck`
- Used both `T1w` and `T2w` images that were aligned beforehand

![image](https://github.com/simnibs/simnibs/assets/9975260/f76d14d8-1783-4fe5-a33d-2ccfe0273d9c)

Why could something like happen? Did anybody else face the same? 

Thank you!

**Comment by oulap** [comment]
There are a few differences between the FS and charm volume segmentations. Also the segmentation approach is different so it's not surprising that there are some differences. I'm not sure what the CSF label in your plot is for example, but FreeSurfer does not segment the cortical CSF whereas charm does. If I remember correctly, the CSF in FS is some small blob of CSF somewhere deep in the brain. I think volume differences around 10% are not unexpected. Higher ones indicate that one of the segmentations might have some errors, e.g., sub HCA6047359 looks like there's probably something going on there. charm is not so good at handling subjects with atrophy right out-of-the-box but the settings can be changed to improve the segmentations, see e.g. this discussion #131 

Also if it's important for you to exactly match the charm and FS brain segmentations, you can stitch the two labelings together and mesh that directly. In this case you would have to relabel the FS neuroanatomical structures to WM and GM so that those can be mapped to conductivities in the simulation.
